add local eval flag configs api and bootstrapping docs (#663)
amplitude/amplitude-docs: content/collections/experiment-apis/en/experiment-evaluation-api.md

source: 'https://www.docs.developers.amplitude.com/experiment/apis/evaluation-api/'
summary:  Lets you retrieve variant assignment data for users with remote evaluation.
---

> The Amplitude Experiment Evaluation APIs retrieve variant assignment data for users through [remote evaluation](/docs/feature-experiment/remote-evaluation) via the [evaluation API](#evaluation-api), or download local evaluation flags using the [flags API](#flags-api).

Reviewer: ```suggestion
The Amplitude Experiment Evaluation APIs retrieve variant assignment data for users through [remote evaluation](/docs/feature-experiment/remote-evaluation) using the [evaluation API](#evaluation-api), or download local evaluation flags using the [flags API](#flags-api).
```

source: 'https://www.docs.developers.amplitude.com/experiment/apis/evaluation-api/'
summary:  Lets you retrieve variant assignment data for users with remote evaluation.
---

> Amplitude Experiment's evaluation APIs lets your re

Reviewer: @bgiori Not sure what happened to this line
```suggestion
```

> The Amplitude Experiment Evaluation APIs lets you retrieve variant assignment data for users via [remote evaluation](/docs/feature-experiment/remote-evaluation) using the evaluation API, or download local evaluation. User information passes as query parameters on the request to allow for [caching the response on the CDN](/docs/feature-experiment/under-the-hood/performance-and-caching#cdn-caching). When you call this API, Amplitude tracks an `[Experiment] Assignment` event.

Reviewer: ```suggestion
The Amplitude Experiment Evaluation APIs retrieve variant assignment data for users through [remote evaluation](/docs/feature-experiment/remote-evaluation) with the evaluation API, or download local evaluation. User information passes as query parameters on the request to allow for [caching the response on the CDN](/docs/feature-experiment/under-the-hood/performance-and-caching#cdn-caching). When you call this API, Amplitude tracks an `[Experiment] Assignment` event.
```

what does `download local evaluation` mean here?

Instead of retrieving remote evaluation from Amplitude servers, you can download the variant assignments that happened locally?

## Regions

| Region | URL |
| ------ | --- |
| Standard Server | `https://api.lab.amplitude.com/v1/` |
| EU Residency Server | `https://api.lab.eu.amplitude.com/v1/` |

## Authorization

The REST API authenticates the request using your [deployment](/docs/feature-experiment/data-model#deployments) key set in the Authorization header with the prefix `Api-Key`. For example, `Authorization: Api-Key <deployment_key>`

---

## Variants API

```text
GET /v1/vardata
```

The Evaluation API lets you retrieve variant assignment data for users.

### Query parameters

| Name | Description |
| ---- | ----------- |
| `user_id` | The user's ID. |
| `device_id` | The user's device ID. |

> | `flag_keys` | Specific flag keys to get the variants of. Multiple flag keys should be separated by commas, e.g. `flag_keys=flag-A,flag-B`. If empty/missing, Experiment evaluates all flags & experiments associated with the deployment key. |

Reviewer: ```suggestion
| `flag_keys` | Specific flag keys to get the variants of. If you have more than one flag key, separate them with commas, for example, `flag_keys=flag-A,flag-B`. If empty/missing, Experiment evaluates all flags & experiments associated with the deployment key. |
```

```json
{}
```

{{/partial:admonition}}

#### Evaluation response status codes

| Status Code | Description |
| ----------- | ----------- |
| 200  | A successful request returns a `200` response and a map of flag key to variants. If `flag_keys` isn't provided, Experiment evaluates all flags associated with the deployment key in the authorization header. Use the [example](#evaluation-example) below to try the API from your browser or copy a curl. |
| 400 | If the request has invalid JSON in the context parameter, it returns a `400` status. |
| 401 | If the request doesn't include a valid API key, it returns a `401` response. |

### Evaluation example

Set the fields in the table, and press send to send the request in browser, or copy the curl to send the request yourself.

{{partial:experiment/interactive-evaluation-table}}

---

## Flags API

```text
GET /v1/flags
```

The Flags API allows you to download flag configurations for local evaluations. This is useful for bootstrapping client-side local evaluation SDKs with `initialFlags` or running evaluation using external flag configuration storage like edge environments.

> The data model returned by this API is different from what you may see in the [management API](./experiment-management-api.md) for flags and experiments. This format is specifically used for evaluation purposes, which is more verbose and less human readable than the management API.

Reviewer: ```suggestion
The data model returned by this API is different from what you may see in the [management API](/docs/apis/experiment/experiment-management-api) for flags and experiments. This format is specifically used for evaluation purposes, which is more verbose and less human readable than the management API.
```

```json
{}
```

{{/partial:admonition}}

#### Evaluation response status codes

| Status Code | Description |
| ----------- | ----------- |
| 200  | A successful request returns a `200` response and a map of flag key to variants. If `flag_keys` isn't provided, Experiment evaluates all flags associated with the deployment key in the authorization header. Use the [example](#evaluation-example) below to try the API from your browser or copy a curl. |
| 400 | If the request has invalid JSON in the context parameter, it returns a `400` status. |
| 401 | If the request doesn't include a valid API key, it returns a `401` response. |

> ### Evaluation Example

Reviewer: ```suggestion
### Evaluation example
```

Set the fields in the table, and press send to send the request in browser, or copy the curl to send the request yourself.

{{partial:experiment/interactive-evaluation-table}}

---

## Flags API

```text
GET /v1/flags
```

The Flags API allows you to download flag configurations for local evaluations. This is useful for bootstrapping client-side local evaluation SDKs with `initialFlags` or running evaluation using external flag configuration storage like edge environments.

The data model returned by this API is different from what you may see in the [management API](./experiment-management-api.md) for flags and experiments. This format is specifically used for evaluation purposes, which is more verbose and less human readable than the management API.

### Flags query parameters

| Name | Description |
| ---- | ----------- |

> | `flag_keys` | Specific flag keys to get the flag configurations for. Multiple flag keys should be separated by commas, e.g. `flag_keys=flag-A,flag-B`. If empty/missing, the API returns all flags & experiments associated with the deployment key. |

Reviewer: ```suggestion
| `flag_keys` | Specific flag keys to get the flag configurations for. If you have more than one flag key, separate them with commas, for example,  `flag_keys=flag-A,flag-B`. If empty/missing, the API returns all flags & experiments associated with the deployment key. |
```
